src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm

```diff
@@ -102,6 +103,8 @@ static void writeAdapter(ExecState* execState, const JSValue& value, void* buffe
         this->_ffiTypeMethodTable.write = &writeAdapter<TNSArrayAdapter>;
     } else if (klass == [NSDictionary class]) {
         this->_ffiTypeMethodTable.write = &writeAdapter<TNSDictionaryAdapter>;
+    } else if (klass == [NSData class]) {
```

Comment:
I'll add a condition for `NSMutableData` as well, because `TNSDataAdapter` actually derives from `NSMutableData`. I don't think using `isSubclassOfClass` will work, though, because there are other descendants of `NSData`.